libyang/data.py

```diff
@@ -954,11 +955,17 @@ def cdata_leaf_value(cdata) -> Any:
 
         if ret in (lib.LY_SUCCESS, lib.LY_EINCOMPLETE):
             val_type = val_type[0].basetype
-            if val_type == Type.BOOL:
-                return val == "true"
+            if val_type in Type.STR_TYPES:
+                return val
             if val_type in Type.NUM_TYPES:
                 return int(val)
-            return val
+            if val_type == Type.BOOL:
+                return val == "true"
+            if val_type == Type.DEC64:
+                return float(val)
+            if val_type == Type.LEAFREF:
+                return DLeaf.cdata_leaf_value(cdata.value.leafref, context)
+            return None
```

Comment:
I wonder if returning `None` instead of a string containing the value is a good idea. How about `identity-ref`, `instance-id` typed nodes?